modules/oe_webtools_analytics/src/EventSubscriber/AnalyticsEventSubscriber.php

```diff
@@ -51,11 +59,14 @@ class AnalyticsEventSubscriber implements EventSubscriberInterface {
    *   The configuration object.
    * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
    *   The request on the stack.
+   * @param \LoggerChannelFactoryInterface $loggerFactory
+   *   The logger channel factory.
    */
-  public function __construct(ConfigFactoryInterface $configFactory, RequestStack $requestStack) {
+  public function __construct(ConfigFactoryInterface $configFactory, RequestStack $requestStack, LoggerChannelFactoryInterface $loggerFactory) {
     // Get id from settings.php!
     $this->config = $configFactory->get(AnalyticsEventInterface::CONFIG_NAME);
     $this->requestStack = $requestStack;
+    $this->loggerFactory = $loggerFactory;
```

Comment:
I would not set logger factory in the class, but directly the channel.

```
$this->logger = $loggerFactory->get('oe_webtools');
```

And then you have `$logger` everywhere you need to log.